enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+
+## Summary
+
+There are several internal pipelines that deal with OLM operator data, all of which need to build or consume the newer bundle and registry formats. We can build new bundle and registry formats as per the other 4.4 epics, but if no one internally is using them we will not derive much of the desired value from them.
+
+### Goals
+
+* Maintainers of Operator Registries can leverage container registries so that they can host Operator bundles and indexes of all available bundles in this catalog in the form of container images
+* Maintainers of Operator Registries can use a tool to:
+    * create a bundle index so that it is easy to employ in a CI environment.
+    * migrate to and from both old and new format.
+    * merge indexes
+    * delete the latest version of an operator from an index
+    * download the contents of index images
+    * specify related images for CVE updates.
+    * validate the contents of an index before publishing it.
+    * package/release/version tool(s)
+* Pipeline admins can publish multiple applications simultaneously without conflicts.
+* Developers or QE can use downstream images in a dev/stage/qe repository
+* There is sufficient documentation for operator owners and maintainers of operatory registries.
+* Design the migration of Community Operators to the new bundle format.
+
+### Non-Goals
+
+* Migrate Community Operators / operatorhub.io to the new bundle format.
+* Pipeline admins rely on the tooling to extrapolate defined selective metadata from full set of each application operator manifests and include that in a combined catalog for index object
+* Pipeline admins can update official catalog object with newly created single application operator and publish
+
+---
+
+## User Stories
+
+### Migrate to and from each registry format
+
+As an Operator Registry maintainer, I want to continue to push updates to old (appregistry-based) catalogs after having migrated to the new bundle format without having to maintain two repositories.
+
+#### Design Proposal
+
+Add a command to `opm index` and `opm alpha bundle` called `migrate` which:
+
+* takes a collection of bundles migrates these to the old manifest format.
+
+    **Example CLI:**
+    `opm alpha bundle migrate --bundles quay.io/my/bundle:v1,quay.io/my/bundle:v2 --out path/to/my/dir`
+    
+* takes an index and migrate it to the old manifest format.
+    
+    **Example CLI:**
+    `opm index migrate quay.io/my/index --out path/to/my/dir`
+    
+    where `path/to/my/dir` would look like:
+    
+    ```
+    dir
+    ├── bundle
+    │   ├── v1
+    │   │   ├── some.crd.yaml
+    │   │   └── bundle.clusterserviceversion.yaml
+    │   ├── v2
+    │   │   ├── some.crd.yaml
+    │   │   ├── some.other.crd.yaml
+    │   │   ├── bundle.v2.clusterserviceversion.yaml
+    │   └── bundle.package.yaml
+    ```
+
+* takes a manifest directory for a given operator and migrates it to an individual bundle
```

Comment:
Is this something that we are expecting to run in CI? Wouldn't this just be something that is manually done by users as needed? Or are we suggesting that we will construct a tool that will iterate over a set of bundles one at a time to create a set of dockerfiles and then use those dockerfiles to publish the bundle images?

If this is a one time operation, my thought is that this isn't something that CI should be doing. Do they have a workflow in mind already?